pkg/crc/machine/status.go

```diff
@@ -33,45 +33,53 @@ func (client *client) Status() (*types.ClusterStatusResult, error) {
 		return nil, errors.Wrap(err, "Cannot get machine state")
 	}
 
+	ip, err := vm.IP()
+	if err != nil {
+		return nil, errors.Wrap(err, "Error getting ip")
+	}
+	ramSize, ramUse := client.getRAMStatus(vm)
+	diskSize, diskUse := client.getDiskDetails(vm)
+	pvSize, pvUse := client.getPVCSize(vm)
+	var openShiftStatusSupplier = getOpenShiftStatus
+	if vm.bundle.IsMicroshift() {
+		openShiftStatusSupplier = getMicroShiftStatus
+	}
+
+	return createClusterStatusResult(vmStatus, vm.bundle.GetBundleType(), vm.bundle.GetVersion(), ip, ramSize, ramUse, diskSize, diskUse, pvSize, pvUse, openShiftStatusSupplier)
+}
+
+func createClusterStatusResult(vmStatus state.State, bundleType preset.Preset, vmBundleVersion string, vmIP string,
+	diskSize int64, diskUse int64, ramSize int64, ramUse int64, pvUse int, pvSize int,
+	openShiftStatusSupplier func(context.Context, string) types.OpenshiftStatus) (*types.ClusterStatusResult, error) {
```

Comment:
nit: maybe you can reduce the length of this line a bit by doing the following, and create a type for the _openShiftStatusSupplier_ func:

```go
type openShiftStatusSupplierFunc func(context.Context, string) types.OpenshiftStatus
```

```suggestion
func createClusterStatusResult(vmStatus state.State, bundleType preset.Preset, vmBundleVersion, vmIP string, diskSize, diskUse, ramSize, ramUse int64, pvUse, pvSize int, openShiftStatusSupplierFunc) (*types.ClusterStatusResult, error) {
```